app/shared-hub/utils/index.js

```diff
@@ -26,6 +26,10 @@ import { withRouter } from 'react-router-dom';
 import { log } from '../../../src/utils/common';
 import { sendMessage as importedSM, sendMessageInPromise as importedSMIP } from '../../panel/utils/msg';
 
+// Self import this file so makeDeferredDispatcher can mock the sendMessageInPromise function within the same class
+// eslint-disable-next-line import/no-self-import
+import * as utils from './index';
+
```

Comment:
Ah, so that's the way to do it. I mocked both `makeDeferredDispatcher` and `sendMessageInPromise` inside the files that need it. Thanks for the help, it was a very tricky bug